### PR TITLE
look for app sources in correct locations

### PR DESCRIPTION
**Pull request by adinn**
Correction to ensure app source roots are computed correctly relative to a jar or directory in the classpath.

### CODE REVIEW

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/sources/ApplicationSourceCache.java [hunk 1]
```diff
@@ -107,6 +107,17 @@ private void trySourceRoot(String sourceRoot, boolean fromClassPath) {
                 // try the path as provided
                 File file = sourcePath.toFile();
                 if (file.exists() && file.isDirectory()) {
+                    // see if we have src/main/java or src/java
+                    Path subPath = sourcePath.resolve("main").resolve("java");
+                    file = subPath.toFile();
+                    if (file.exists() && file.isDirectory()) {
+                        sourcePath = subPath;
+                    } else {
+                        subPath = sourcePath.resolve("java");
```
Comment (olpaw) [inline substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/sources/ApplicationSourceCache.java:116]: aren't you missing another
```
file = subPath.toFile();
```
here?

Comment (olpaw) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/sources/ApplicationSourceCache.java:116]: cc @adinn

Comment (adinn) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/sources/ApplicationSourceCache.java:116]: doh! pushed a correction.

Comment (olpaw) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/sources/ApplicationSourceCache.java:116]: hehe ... that's why I prefer Files.isDirectory(...), Files.exists(...), ...

**Comment by adinn** [comment]
Hmm, this seems to have passed all Travis tests despite the indication to the contrary.

**Comment by adinn** [comment]
Ah just as I post that the green light comes up!

**Comment by olpaw** [comment]
Everything looks fine. Our internal gate checks are a bit flaky atm. So it takes a bit of time to get it merged.

**Comment by adinn** [comment]
Sure, apologies for the impatience and thanks for confirming :-)

**Comment by olpaw** [comment]
@adinn your change is on master. Unfortunately 20.1 got already branched. Do you need this change also for the 20.1 release or is it ok to have this land in 20.2?

**Comment by adinn** [comment]
Well, it would be very nice to have :-) However, it is not critical since I don't suppose anyone actually depends on this feature yet.

A reason maybe for including it is that it's going to be hard for anyone to test if it keeps on failing to find source jars without being pointed explicitly at them ... and testing -- even if only by very early adopters -- would be a good idea (I am still intending to produce those scripted tests but ....).

**Comment by olpaw** [comment]
Alright, since it's such a small/local change I will add a backport PR so that we have this also in 20.1.

**Comment by adinn** [comment]
Excellent. Thanks!

**Pull request merged**